### PR TITLE
[PR] Correctly configure greedy switch during hierarchical layout

### PR DESCRIPTION
**Pull request by uruuru**
See #628 for details and discussion.

### CODE REVIEW

File: plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/ElkLayered.java [hunk 1]
```diff
@@ -341,12 +341,11 @@ private void reviewAndCorrectHierarchicalProcessors(final LGraph root, final Col
             });
         }
         
-        // Greedy switch
-        //  simply turn it off for children
-        if (!GraphConfigurator.activateGreedySwitchFor(root)) {
-            graphs.forEach(
-                    g -> g.setProperty(LayeredOptions.CROSSING_MINIMIZATION_GREEDY_SWITCH_TYPE, GreedySwitchType.OFF));
-        }
+        // Greedy switch (simply copy the behavior of the root to all children)
```
Comment (uruuru) [inline plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/ElkLayered.java:344]: From looking at the old implementation and how crossing minimization itself is configured, I feel that every hierarchy level requires the hierarchical-aware layout processor to be configured (even though it won't get executed). 

I didn't check this thoroughly though.

Comment (le-cds) [reply plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/ElkLayered.java:344]: Well, that would make sense to me.

File: plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/Layered.melk [hunk 2]
```diff
@@ -345,14 +346,36 @@ group crossingMinimization {
...
+            description 
                 "Greedy Switch strategy for crossing minimization. The greedy switch heuristic is executed 
-                 after the regular layer sweep as a post-processor."
+                 after the regular crossing minimization as a post-processor. 
+                 Note that if 'hierarchyHandling' is set to 'INCLUDE_CHILDREN', the 'greedySwitchHierarchical.type' 
+                 option must be used."
             default = GreedySwitchType.TWO_SIDED
             targets parents
+            requires crossingMinimization.strategy == CrossingMinimizationStrategy.LAYER_SWEEP
+            requires org.eclipse.elk.hierarchyHandling == HierarchyHandling.SEPARATE_CHILDREN
```
Comment (le-cds) [inline plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/Layered.melk:357]: Will this work for graphs that don't have their hierarchy handling set? Is this the default option?

Comment (uruuru) [reply plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/Layered.melk:357]: In fact, the default is `INHERIT` (which is turned into `SEPARATE_CHILDREN` during `RecursiveGraphLayoutEngine#evaluateHierarchyHandlingInheritance(...)`). 

As a validation of the requires (although I'm not sure if we've something like it implemented right now) would probably happen beforehand, the line would be problematic. 

I guess I'll remove the line. Unfortunately there's no `!=` - which is the task of another long-standing ticket.

Comment (le-cds) [reply plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/Layered.melk:357]: Sounds good.

**Pull request merged**